lib/test/unit/ui/tap/ext/base_testrunner.rb

```diff
@@ -0,0 +1,62 @@
+require 'test/unit/ui/tap/base_testrunner'
+
+module Test
+  module Unit
+    module UI
+      module Tap
+        class Tap::BaseTestRunner < Test::Unit::UI::TestRunner
+
+          def tapout_pass_with_source_location(test)
+            doc = tapout_pass_without_source_location(test)
+            if(doc && doc['type'] == 'test')
+              file, _ = test.class.instance_method(clean_label(test.name).to_sym).source_location
+              doc['file'] = file.sub(Dir.pwd+'/', '')
+            end
+            doc
+          end
+
+          def tapout_todo_with_source_location(fault)
+            doc = tapout_todo(fault)
+            doc['file'] = extract_source_location
```

Comment:
@shiroyasha The `tapout_pass` method returns a nil if a test was already processed with a fault.  That means, it was already handled by one of the `tapout_(error| fail | todo| omit)` methods.  Here is the [https://github.com/renderedtext/test-unit-runner-tap/blob/master/lib/test/unit/ui/tap/base_testrunner.rb#L159|code snippet]. 

The rest of the methods always return a `doc`, hence I did not add check for existence of `doc`.